Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -493,6 +494,17 @@ StyleAppearance RenderTheme::autoAppearanceForElement(RenderStyle& style, const
 
         if (pseudo == ShadowPseudoIds::webkitInnerSpinButton())
             return StyleAppearance::InnerSpinButton;
+
+        if (auto host = element->shadowHost()) {
+            if (host->computedStyle()->effectiveAppearance() != StyleAppearance::Auto)
```

Comment:
I tested this and this works well, but I'm not familiar with these internal APIs and whether it's okay to invoke them here.

Reply:
I wouldn't use `!important`, but inherit could be a good idea. It would need some care if there's a complex shadow structure though.

Reply:
I think `!important` makes sense as it ensures you cannot get half of the UA styling. Either all or nothing.